### PR TITLE
Documentation changes for new SQL support

### PR DESCRIPTION
**Pull request by mparkhe**
`mlflow ui` and `mlflow server` now have a new option `--backend-store-uri`, with old `--file-store` preserved as an alias.

Additionally, MLflow tracking server now supports SQL backed backends.

### CODE REVIEW

File: docs/source/tracking.rst [hunk 1]
```diff
@@ -57,7 +57,8 @@ API.
 Where Runs Are Recorded
 =======================
 
-MLflow runs are recorded either locally in files or remotely to a tracking server.
+MLflow runs are recorded either locally in files, or SQLAlchemy compatible database, or remotely
```
Comment (dbczumar) [inline docs/source/tracking.rst]: How do you feel about using a bulleted list? The phrasing here is a little hard to parse.

```
Users can record MLflow runs in the following locations:

- **Locally** as files
- **Locally** as entries in a SQLAlchemy-compatible database
- **Remotely** to a tracking server
```

[the remark this answers is not in the document]
Comment (stbof) [reply docs/source/tracking.rst]: What about: MLflow runs can be recorded to local files, to a SQLAlchemy compatible database, or to a tracking server.

File: docs/source/tracking.rst [hunk 2]
```diff
@@ -83,8 +85,8 @@ Basic Logging Functions
 
 :py:func:`mlflow.set_tracking_uri` connects to a tracking URI. You can also set the
 ``MLFLOW_TRACKING_URI`` environment variable to have MLflow find a URI from there. In both cases,
-the URI can either be a HTTP/HTTPS URI for a remote server, or a local path to log data to a
-directory. The URI defaults to ``mlruns``.
+the URI can either be a HTTP/HTTPS URI for a remote server, or, database connection string, or a
```
Comment (dbczumar) [inline docs/source/tracking.rst]: I think the following is clearer:

```
the URI can either be a HTTP/HTTPS URI for a remote server, a database connection string, or a local directory path.
```

**Review by smurching — APPROVED** [review]
LGTM thanks @mparkhe
